scenes/game/game.gd

```diff
@@ -4424,152 +4401,55 @@ func card_in_selected_cards(card):
-	# Gauges
-	await _update_popout_cards(
-		$AllCards/PlayerGauge.get_children(),
-		$PlayerZones/PlayerGauge.get_center_pos(),
-		CardBase.CardState.CardState_InGauge
-	)
-	await _update_popout_cards(
-		$AllCards/OpponentGauge.get_children(),
-		$OpponentZones/OpponentGauge.get_center_pos(),
-		CardBase.CardState.CardState_InGauge
-	)
-
-	# Sealed area
-	await _update_popout_cards(
-		$AllCards/PlayerSealed.get_children(),
-		$PlayerZones/PlayerSealed.get_center_pos(),
-		CardBase.CardState.CardState_InGauge
-	)
-	await _update_popout_cards(
-		$AllCards/OpponentSealed.get_children(),
-		$OpponentZones/OpponentSealed.get_center_pos(),
-		CardBase.CardState.CardState_InGauge
-	)
-
-	# Overdrive area
-	await _update_popout_cards(
-		$AllCards/PlayerOverdrive.get_children(),
-		$PlayerZones/PlayerOverdrive.get_center_pos(),
-		CardBase.CardState.CardState_InGauge
-	)
-	await _update_popout_cards(
-		$AllCards/OpponentOverdrive.get_children(),
-		$OpponentZones/OpponentOverdrive.get_center_pos(),
-		CardBase.CardState.CardState_InGauge
-	)
 
-	# Discards
-	await _update_popout_cards(
-		$AllCards/PlayerDiscards.get_children(),
-		get_discard_location($PlayerDeck/Discard),
-		CardBase.CardState.CardState_Discarded
-	)
-	await _update_popout_cards(
-		$AllCards/OpponentDiscards.get_children(),
-		get_discard_location($OpponentDeck/Discard),
-		CardBase.CardState.CardState_Discarded
-	)
-
-	# Boosts
-	await _update_popout_cards(
-		$AllCards/PlayerBoosts.get_children(),
-		get_boost_zone_center($PlayerBoostZone),
-		CardBase.CardState.CardState_InBoost
-	)
-	await _update_popout_cards(
-		$AllCards/OpponentBoosts.get_children(),
-		get_boost_zone_center($OpponentBoostZone),
-		CardBase.CardState.CardState_InBoost
-	)
-
-	# Reference
-	await _update_popout_cards(
-		$AllCards/PlayerAllCopy.get_children(),
-		OffScreen,
-		CardBase.CardState.CardState_Offscreen
-	)
-	await _update_popout_cards(
-		$AllCards/OpponentAllCopy.get_children(),
-		OffScreen,
-		CardBase.CardState.CardState_Offscreen
-	)
-
-	# Revealed
-	await _update_popout_cards(
-		$AllCards/OpponentRevealed.get_children(),
-		OffScreen,
-		CardBase.CardState.CardState_Offscreen
-	)
+	var cards = []
+	for card in cards_in_popout:
+		if filtering_allowed and popout_show_normal_only() and not game_wrapper.get_card_database().is_normal_card(card.card_id - ReferenceScreenIdRangeStart):
+			continue
+		cards.append(card)
 
-	# Choice Zone
-	await _update_popout_cards(
-		$AllCards/ChoiceZone.get_children(),
-		OffScreen,
-		CardBase.CardState.CardState_InDeck
-	)
+	card_popout.show_cards(cards)
+	for card in selected_cards:
+		card_popout.modify_card_selection(card.card_id, true)
 
 func close_popout():
-	card_popout.visible = false
-	if popout_showing_node:
-		popout_showing_node.z_index = 0
-	await clear_card_popout()
+	while card_popout_parent.get_child_count() > 0:
+		var child = card_popout_parent.get_child(0)
+		card_popout_parent.remove_child(child)
+		child.queue_free()
```

Comment:
From a quick search of what queue_free does, it seems like the remove_child call might be redundant; have you tried seeing if anything changes if you delete that call?

Reply:
even though queue_free does remove the node from the scene tree, I want to be super explicit about removing it immediately here because i'm also adding a child next and referring to child 0, so I just want to be very sure I'm referring to the new one and not the old one.